cmake/CMakeLists.txt

```diff
@@ -2157,65 +1365,109 @@ if (onnxruntime_USE_WINML)
 endif() # if (onnxruntime_USE_WINML)
 
-
-#The following files may use the 'onnxruntime_libs' and 'onnxruntime_EXTERNAL_LIBRARIES' vars
-
 if (onnxruntime_BUILD_SHARED_LIB OR onnxruntime_BUILD_APPLE_FRAMEWORK)
   if (onnxruntime_BUILD_APPLE_FRAMEWORK AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin|iOS")
     message(FATAL_ERROR "onnxruntime_BUILD_APPLE_FRAMEWORK can only be enabled for macOS or iOS.")
   endif()
-  include(onnxruntime.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime)
 endif()
 
 if (onnxruntime_BUILD_JAVA)
   message(STATUS "Java Build is enabled")
-  include(onnxruntime_java.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_java)
 endif()
 
 if (onnxruntime_BUILD_NODEJS)
   message(STATUS "Node.js Build is enabled")
-  include(onnxruntime_nodejs.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_nodejs)
 endif()
 
 if (onnxruntime_ENABLE_PYTHON)
-  include(onnxruntime_python.cmake)
+  message(STATUS "Python Build is enabled")
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_python)
 endif()
 
 if (onnxruntime_BUILD_OBJC)
   message(STATUS "Objective-C Build is enabled")
-  include(onnxruntime_objectivec.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_objectivec)
 endif()
 
 if (onnxruntime_BUILD_UNIT_TESTS)
-  include(onnxruntime_unittests.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_unittests)
+endif()
+
+if (onnxruntime_BUILD_WINML_TESTS)
+  list(APPEND ONNXRUNTIME_TARGETS winml_unittests)
 endif()
 
+# onnxruntime_training depends on onnxruntime_unittests since onnxruntime_training.cmake uses a variable `TEST_SRC_DIR`
+# that is defined in onnxruntime_unittests.cmake
 if (onnxruntime_ENABLE_TRAINING)
-  include(onnxruntime_training.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_training)
   if (onnxruntime_ENABLE_TRAINING_E2E_TESTS)
-    include(onnxruntime_training_e2e_tests.cmake)
+    list(APPEND ONNXRUNTIME_TARGETS onnxruntime_training_e2e_tests)
   endif()
 endif()
 
 if (onnxruntime_BUILD_CSHARP)
   message(STATUS "CSharp Build is enabled")
 #  set_property(GLOBAL PROPERTY VS_DOTNET_TARGET_FRAMEWORK_VERSION "netstandard2.0")
-  include(onnxruntime_csharp.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_csharp)
 endif()
 
 if (onnxruntime_BUILD_WEBASSEMBLY)
   message(STATUS "WebAssembly Build is enabled")
-  include(onnxruntime_webassembly.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_webassembly)
 endif()
 
 if(onnxruntime_BUILD_KERNEL_EXPLORER)
   message(STATUS "Kernel Explorer Build is enabled")
-  include(onnxruntime_kernel_explorer.cmake)
+  list(APPEND ONNXRUNTIME_TARGETS onnxruntime_kernel_explorer)
+endif()
+
+if (onnxruntime_ENABLE_EAGER_MODE)
+  if (NOT onnxruntime_ENABLE_TRAINING OR NOT onnxruntime_ENABLE_PYTHON)
+    message(
+        FATAL_ERROR
+        "Option onnxruntime_ENABLE_EAGER_MODE can only be used when onnxruntime_ENABLE_TRAINING and onnxruntime_ENABLE_PYTHON are enabled")
+  endif()
+  add_compile_definitions(ENABLE_EAGER_MODE)
```

Comment:
should this go into adjust_global_compile_flags.cmake?

Reply:
It doesn't need to be applied to external dependencies. So I would prefer to put it in onnxruntime_set_compile_flags